recipes/full_finetune_distributed.py

```diff
@@ -681,7 +727,13 @@ def train(self) -> None:
                         time_per_step = time.perf_counter() - t0
                         log_dict = {
                             "loss": loss_to_log,
-                            "lr": self._optimizer.param_groups[0]["lr"],
+                            "lr": (
+                                list(self._optim_ckpt_wrapper.values())[0].param_groups[
```

Comment:
make this a util function? probably better to assert on "lr" is the same across param_groups

Reply:
add get_lr_scheduler to check whether all the lr_scheduler are the same and return it